File: fitbud-backend/src/main/java/com/example/fitbudbackend/workout/entities/WorkoutExercise.java
package com.example.fitbudbackend.workout.entities;

import jakarta.persistence.*;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;

@Entity
@Table(name = "workout_exercises")
@Data
@NoArgsConstructor
@AllArgsConstructor
@Builder
public class WorkoutExercise {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "workout_day_id", nullable = false)
    private WorkoutDay workoutDay;

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "exercise_id")
    private Exercise exercise;

    private String customName;

    private Integer sets;

    private Integer reps;

    private Double weight;

    private Integer exerciseOrder;

}

File: fitbud-frontend/src/features/workout/types/WorkoutExercise.ts
export interface WorkoutExercise {
    id: number;
    exerciseId: number | null;
    exerciseName: string | null;
    emoji: string | null;
    customName: string | null;
    sets: number;
    reps: number;
    weight: number;
    exerciseOrder: number;
}